### ISSUE TITLE
Could you help remove the high severity vulnerabilities introduced in your package?

### ISSUE THREAD

**Issue by paimon0715** [Automattic/mongoose#10484]
Hi ,@vkarpov15, @AbdelrahmanHafez , I’d like to report two vulnerabilities introduced in **mongoose** :
### Issue Description
   Two vulnerabilities (high severity) [**CVE-2019-2391**](https://snyk.io/vuln/SNYK-JS-BSON-561052) and [**CVE-2020-7610**](https://snyk.io/vuln/SNYK-JS-BSON-561052) are detected in package **bson(>=1.0.0 <1.1.4)** and **bson@1.0.9** is directly referenced by **_mongoose@4.13.21_**. We noticed that the vulnerabilities has been removed since _**mongoose@5.3.9**_.
  
   However, mongoose's popular previous version **_mongoose@4.13.21_**  ([25,347 downloads per week](https://www.npmjs.com/package/mongoose/v/4.13.21?activeTab=versions)) is still transitively referenced by a large amount of latest versions of active and popular downstream projects (about **3,324 downstream projects**, e.g., @app-masters/node-lib 2.2.1, omniboard 2.14.0, e-commerce-platform 0.0.1, capstonejs 4.2.23-b, @shoutem/express-stack 0.2.36, etc.).
As such, issue [**CVE-2019-2391**](https://snyk.io/vuln/SNYK-JS-BSON-561052) and [**CVE-2020-7610**](https://snyk.io/vuln/SNYK-JS-BSON-561052) can be propagated into these downstream projects and expose security threats to them.
   
   These projects cannot easily upgrade **mongoose** from version **_8.13.0_** to _**(>=9.2.0)**_ For instance, **_mongoose@4.13.21_** is introduced into the above projects via the following package dependency paths:
(1)`@app-masters/node-lib@2.2.1 ➔ @app-masters/mongoose-it@1.0.13 ➔ mongoose@4.13.21 ➔ bson@1.0.9`
    **......**
    
The projects such as **@app-masters/mongoose-it**, which introduced mongoose@4.13.21, are not maintained anymore. These unmaintained packages can neither upgrade **mongoose** nor be easily migrated by the large amount of affected downstream projects.  
 On behalf the downstream users, could you help us remove the vulnerabilities from package mongoose@4.13.21?

### Suggested Solution

Since these unactive projects set a version constaint **4.13.\*** for **mongoose** on the above vulnerable dependency paths, if **mongoose** removes the vulnerability from 4.13.21 and releases a new patched version **_mongoose@4.13.22_**, such a vulnerability patch can be automatically propagated into the 3,324 affected downstream projects.

In **_mongoose@4.13.22_**, you can kindly try to perform the following upgrade:
       `bson ~1.0.4 ➔ ~ 1.1.4`;  
**Note:**
**_bson@1.1.4(>=1.1.4)_** has fixed the vulnerabilityies ([**CVE-2019-2391**](https://snyk.io/vuln/SNYK-JS-BSON-561052) and [**CVE-2020-7610**](https://snyk.io/vuln/SNYK-JS-BSON-561052))

Thank you for your contributions.

Yours sincerely,
Paimon

**Comment by paimon0715** [comment]
@AbdelrahmanHafez @IslandRhythms Thanks for your understanding and help.

**Comment by vkarpov15** [comment]
Copying my comment from #10489 here for visibility for @paimon0715 :

I took a closer look at this and this change won't help. That's because Mongoose 4.x [relies on MongoDB driver 2.x](https://github.com/Automattic/mongoose/blob/d63d08cff633892e67da4b9597a807ecc921a6dd/package.json#L27), which [relies on mongodb-core@2.1.x](https://github.com/mongodb/node-mongodb-native/blob/110ecd07a6f69cab79364b0b7226f99ebff408f5/package.json#L17), which [relies on `bson@~1.0.4`](https://github.com/mongodb-js/mongodb-core/blob/e5991009d7e796c41e5aa529deba464988215200/package.json#L21). So this PR won't help unless we get https://github.com/mongodb-js/mongodb-core/pull/464 merged.

TLDR; Mongoose can't do anything about this without the MongoDB driver team's help. Is there anything we can do to help you upgrade to Mongoose 5?

**Comment by levpachmanov** [comment]
Hey @paimon0715 ,
We're part of a startup called Seal Security that mitigates software vulnerabilities in older open source versions by backporting/creating standalone security patches - enabling more straightforward remediation in cases like this. We created an `bson@1.0.9-sp1` that's vulnerability-free. As with all of our patches, it's open-source and available for free.

If relevant, check out [our GitHub repo](https://github.com/seal-community/patches/blob/main/npm/bson/1.0.9/sp1/0001-CVE-2020-7610-and-more.patch) if you wish to learn more, or start using our [app](https://app.sealsecurity.io/?ref=gh-bson) - it's free to use for open-source projects!

Please feel free to reach us at [info@seal.security](mailto:info@seal.security) if you have any requests/questions.
